terraform/main.tf

```diff
@@ -2,15 +2,40 @@
 # See LICENSE file for licensing details.
 
 resource "juju_application" "mongodb" {
-  name  = var.app_name
-  model = var.model_name
 
   charm {
-    name    = "mongodb"
-    channel = var.channel
-    base    = "ubuntu@22.04"
+    name     = "mongodb"
+    channel  = var.channel
+    revision = var.revision
+    base     = var.base
```

Comment:
Should we enforce the base for now since I'm not sure we're supporting more recent bases?

Reply:
Good question! I have removed it for now